feat #5 : 회원가입 기능

sproutboot opened the pull request:
- 유저 엔티티 설계
- spring validation 적용(필수 입력값 지정, 비밀번호 정규 표현식)
- 유저 아이디 중복체크 처리
- 유저 비밀번호 Bcrypt 암호화를 통해 저장
- 회원가입 기능 구현
- 회원가입 기능 테스트

sproutboot committed:
feat #5 : 회원가입 기능
- 유저 엔티티 설계
- spring validation 적용(필수 입력값 지정, 비밀번호 정규 표현식)
- 유저 아이디 중복체크 처리
- 유저 비밀번호 Bcrypt 암호화를 통해 저장
- 회원가입 기능 구현
- 회원가입 기능 테스트

f-lab-joseph commented on src/test/java/com/wants/market/user/service/UserServiceImplTest.java:
```diff
@@ -0,0 +1,63 @@
+        try {
+            userService.signUp(createUserRequest);
+        } catch(Exception e) {
+            System.out.println("오류 발생");
```
signup 이 성공하더라도 테스트는 성공으로 표기될 것 같습니다.

sproutboot committed:
feat #5 : 회원가입 기능 테스트
- spock 회원가입 기능 테스트
- 존재하지 않는 이메일로 가입 시도 성공
- 존재하는 이메일로 가입 시도시 DuplicatedIdException 발생
